src/dmd/e2ir.d

```diff
@@ -659,8 +664,19 @@ elem *array_toPtr(Type t, elem *e)
             }
             else if (e.Eoper == OPpair)
             {
-                e.Eoper = OPcomma;
-                e.Ety = TYnptr;
+                if (el_sideeffect(e.EV.E1))
+                {
+                    e.Eoper = OPcomma;
+                    e.Ety = TYnptr;
+                }
+                else
+                {
+                    auto r = e;
+                    e = e.EV.E2;
+                    e.Ety = TYnptr;
+                    r.EV.E2 = null;
+                    el_free(r);
```

Comment:
Note: it appears that this branch skips evaluating the length of the array and just returns the array's pointer directly.